### PR TITLE
update `halo2curves` to 0.6.0

### PR DESCRIPTION
**Pull request by kilic**
Apperently this update won't require zkcrypto changes. 

* Bounds between `pairing::Engine` and `{CurveAffine, CurveExt}` are satisfied
* It wasn't compiling on my end because of `ilog2` so I bumped the toolchain to 1.67.0 from 1.66.0. But then lots of new clippy warnings are appeared. So should I rollback to 1.66 or fix clippy also here in this PR?

Question: should we bump halo2_proofs version to 0.2.1?

It should replace #234

### CODE REVIEW

**Review by CPerezz — APPROVED** [review]
LGTM modulo tests passing.

**Comment by kilic** [comment]
Can you check again if compilation issues fixed correctly? @CPerezz

**Comment by CPerezz** [comment]
> @CPerezz can you check again if compilation issues fixed correctly? Then I'll fix clippy

Looks like it. 
Wasm build/tests aren't working for the same errors than previous. But unsure if it's due architecture target change or what. 
Probably, it makes sense to ignore for now and address later? Or do you want to take a quick look to try to spot the issue?

**Comment by CPerezz** [comment]
@kilic 

This commit: https://github.com/privacy-scaling-explorations/halo2/commit/ab4e2fee848fa6e675d1fc50f77144b0ea9cc9ee

And the changes I introduce in https://github.com/privacy-scaling-explorations/halo2curves/pull/121 should completely fix this for the future.

**Review by han0110 — APPROVED** [review]
LGTM!

**Comment by CPerezz** [comment]
Hey @kilic can you try to see how https://github.com/privacy-scaling-explorations/halo2curves/tree/main now that we merged https://github.com/privacy-scaling-explorations/halo2curves/pull/122??

If works well, I'd rather release a new `halo2curves` version and use it here.

**Comment by kilic** [comment]
> If works well, I'd rather release a new halo2curves version and use it here.

It worked and is waiting for new halo2curves version @CPerezz

**Comment by CPerezz** [comment]
@kilic There it is: https://crates.io/crates/halo2curves/0.6.0

File: halo2_proofs/Cargo.toml
Comment (CPerezz) [inline halo2_proofs/Cargo.toml:83]: Why are removing the `bits` feature? We should bring it back even if it's not as `default`.

**Comment by kilic** [comment]
> I think we could maybe unify now `assert_satisfied_par` and `assert_satisfied` since rayon can internally handle `multicore` by itself?

Also done!

**Comment by CPerezz** [comment]
I think we can merge now @kilic

**Comment by CPerezz** [comment]
Sadly, just after seeing the merge, we got a message from @huitseker which says:
>  have tests hashing public parameters in Supernova on Arecibo, and I noticed the hash of our reference circuits on Bn254/Grumpkin changed when (only) switching from halo2curves 0.5.0 to 0.6.0. This could come from the field ops outputs changing, or the msm output changing, or the serialization format changing. I'll look into this, but: Is that expected? is there a PR I should look at to narrow down what happened? 

So I'll not release until we figure out what happens.

**Pull request merged**
